### ISSUE TITLE
\dots \ldots and \textellipsis at end of \gla is problematic

### ISSUE THREAD

**Issue by cweedall** [frampton/ExPex#6]
When using `\dots` or `\ldots` or `\textellipsis`, at the end of `\gla` (immediately before the `//`) results in no ellipsis/dots displaying and a + appearing at the beginning of the first line of the output.  The problem does not occur when `\dots` or `\ldots` or `\textellipsis` are used at the end of `\glb`.  Furthermore, if you use `\dots` or `\ldots` or `\textellipsis` at the beginning of \gla or in the middle, there's no issue.  This issue can be replicated between `\pex ... \xe` and `\ex ... \xe`.

The problem can be avoided either by using `\expandafter\dots` or `\expandafter\ldots` or `\expandafter\textellipsis` **OR** `\dots\relax` or `\ldots\relax` or `\textellipsis\relax`.  Surrounding the commands by curly braces, such as `{\dots}` or `{\ldots}` or `{\textellipsis}`, or an empty set, such as `\dots{}` or `\ldots{}` or `\textellipsis{}`, (or both), does nothing to mitigate the problem.

This seems wrong to me... and there is nothing in the manual about this type of behavior.  The `\dots` command is used in several places in the manual, but only in the preamble of an example.  So, it seems to me to be a bug...  but I'm unclear on this point.

MWE (problematic with +):
```
\documentclass{article}

\usepackage{expex}

\begin{document}

    \pex
	    \a
		    \begingl
			    \gla original stuff \dots //
			    \glb gloss stuff \dots //
			    \glft `A translation \dots' //
		    \endgl
    \xe

\end{document}
```

MWE (not problematic):
```
\documentclass{article}

\usepackage{expex}

\begin{document}

    \pex
	    \a
		    \begingl
			    \gla original stuff working //
			    \glb gloss stuff \dots //
			    \glft `A translation \dots' //
		    \endgl
    \xe

\end{document}
```

Same behavior with other document classes, such as `memoir`.

I'm opening this as a (bug) issue.  If it is working as expected, then *should* it be expected?  If it will not be updated/fix, then can the manual be updated to explain these (and likely similar situations) and provide a clear suggested method/workaround?

**Comment by cweedall** [comment]
A  similar problem exists with the `xellipsis` package.  The problem is worse, however, because the first line has "0pt ... +" prepended, instead of only +.

MWE: (with xellipsis):
```
\documentclass{article}

\usepackage{expex}
\usepackage[latex]{xellipsis}


\begin{document}

    \pex
	    \a
		    \begingl
			    \gla original stuff \xelip //
			    \glb gloss stuff \dots //
			    \glft `A translation \dots' //
		    \endgl
    \xe

\end{document}
```

Using `\xelipend` from `xellipsis` package further breaks by producing the following error: `! Missing } inserted.`  MWE:
```
\documentclass{article}

\usepackage{expex}
\usepackage[latex]{xellipsis}


\begin{document}

    \pex
	    \a
		    \begingl
			    \gla original stuff \xelipend //
			    \glb gloss stuff \dots //
			    \glft `A translation \dots' //
		    \endgl
    \xe

\end{document}
```

As with the OP, the `\expandafter` and `\relax` methods resolve both of these issues.

**Comment by frampton** [comment]
I will take a careful look in the near future, but it may simply be that {\dots} should replace \dots.  The items on the gla line are space separated, so \gla reads \dots// as a single item .

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by cweedall** [comment]
Sorry for not including a MWE example with it, but in the second paragraph of the OP, I mentioned that `{\dots}` does not work either.  For example, using `\gla original stuff {\dots} //` and `\gla original stuff \dots{} //` in the MWE's above, both result in the same behavior as `\gla original stuff \dots //`.

**Comment by frampton** [comment]
Try putting \dots inside an hbox

\gla xx xx \hbox{\dots} //

Glossing requires a lot of lookahead which can run into problems if the first item encountered is not a character.





Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by cweedall** [comment]
Yes, `\hbox{...}` also appears to resolve the issue for me.

As I mentioned in the OP, I am not entirely sure whether this should be treated as a 'bug' or simply referred to in the manual, along with these 'fixes'.

If there is something that can be done within `expex` to make this seemless to users, I think it may be preferable, personally.  But either option is fine for me.

**Comment by frampton** [comment]
I think I will simply say in the manual that if a gl item begin with an item whose expansion is not character initial, then the item should be put in a box.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by cweedall** [comment]
Ok, a quick follow-up then, because you mentioned 'gl item'.

I do *not* get this behavior for `\glb`.

I'm a bit concerned about your wording suggesting that `\gla` acts the same as `\glb`, whereas they behave differently for me.  I am not entire sure why this is...  Perhaps because the `\glft` line always begins with an apostrophe (i.e. being quote)?

Consider the following MWE which displays the difference (in the attached image):
```
\documentclass{article}

\usepackage{expex}

\begin{document}

    \pex
	    \a
		    \begingl
			    \gla original stuff \dots //
			    \glb gloss stuff \dots //
			    \glft `A translation \dots' //
		    \endgl
		\a
		    \begingl
			    \gla original stuff stuff //
			    \glb gloss stuff \dots //
			    \glft `A translation \dots' //
		    \endgl
    \xe

\end{document}
```
![test2](https://user-images.githubusercontent.com/5010253/33093001-61c6d65a-cec1-11e7-8e0e-5f4b17ccdc9b.jpg)

**Comment by frampton** [comment]
Right.  If you look at the section in the manual “Exceptional \gla items” you will see that the parser must test to see if the item is ‘+’ or ‘@’.  That testing is most likely where the problem is.  That testing does not occur on the other gloss lines.

Perhaps I can modify the code so that it first tests to see if the item is a character, then (if it is) tests to see if it is + or @.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by frampton** [comment]
BTW, your example is not a bad example, but not minimal ---  unless you think a4paper, oneside, fontenc, and fontspec might have something to do with the aberrant behavior.

Same goes for the example and parts labels, which are not relevant.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by cweedall** [comment]
Nope, no reason for the extras stuff other than copying and pasting.  I removed all of the unnecessary packages and labels and it works the same.  I also updated all my previous comments to remove those things.  Sorry to distract from the issue at hand!
